### DISCUSSION TITLE
Yet another menu question

### DISCUSSION THREAD

**Discussion by csparker247** [thuliteio/doks#1024] [Q&A] [answered]
I'm trying to setup an `index.md` file for my custom sections to serve as the natural landing page, but also include a section menu in the left sidebar. Using auto menus and the nested structures, I've successfully setup a menu structure that looks like what I want:

<img width="617" alt="looks-correct" src="https://user-images.githubusercontent.com/1434526/223620139-131a93b5-b184-4587-9034-0e1f5d615cea.png">

The key is that my first level should include the section title, while the link for the first item in the second level points to the root of the section. However, the url for this page is: `DOMAIN/new-imaging-methods/content/introduction/`, which leads to all sorts of shenanigans with links, search, bread crumbs, etc. I would really like to see the above but have the url be `DOMAIN/new-imaging-methods/`.

I thought the solution was to turn off auto-menus and create the menus myself, but this doesn't seem to work:

```toml
# menus.en.toml
[[nim]]
  name = "New imaging methods"
  weight = 10
  identifier = "nim-main"
  url = "/new-imaging-methods/"
```

```markdown
<!--- new-imaging-methods/index.md --->
menu:
  nim:
    parent: "nim-main"
    identifier: "nim-introduction"
    name: "Introduction"
```

<img width="608" alt="correct-structure" src="https://user-images.githubusercontent.com/1434526/223621255-9262f175-7fbd-46b3-9526-2e604a0f9968.png">

Is this even possible? If not, that's fine. I feel like maybe I'm missing something fundamental about Hugo and page bundles. Should I perhaps duplicate the docs layout and modify the list page to look like the single page?

**Comment by h-enk** [comment, marked as answer]
> Should I perhaps duplicate the docs layout and modify the list page to look like the single page?

Yes, that's it. You were almost there 😉 

So, this is I guess the idea:

![Snag_189c46aa](https://user-images.githubusercontent.com/3902872/223658706-aba02738-00d9-4792-8204-a716553a732a.png)

How I got there:

- Copy the contents of `./layouts/docs/single.html` into `./layouts/docs/list.html`, but keep the card list of pages of the current section (I put it directly below `{{ .Content }}`)
-  Add below [line 13 of auto-collapsible-menu.html](https://github.com/h-enk/doks/blob/master/layouts/partials/sidebar/auto-collapsible-menu.html#L13)

```html
            {{ $active := eq $currentPage.RelPermalink .RelPermalink }}
            <li><a class="docs-link rounded{{ if $active }} active{{ end }}" href="{{ .Permalink }}">Overview</a></li>
```

Hope this helps you get it your way

**Reply by csparker247** [reply]
Alright! That got me much closer to what I wanted! 

<img width="807" alt="corrected" src="https://user-images.githubusercontent.com/1434526/223888831-097ae8d4-5611-4a40-8c0d-2a680514ad36.png">

For completeness, I was actually using default menus (not collapsible), so I added your lines just under the start of the `<ul>` in `auto-default-menu.html`. Also, using `href="{{ .Permalink }}` for my landing page menu entry resulted in a relative link of `section/subdirectory/` when I just needed it to be section. I managed to get around this by making it `href="/{{ .Section }}/"`. It feels like this could be more elegant though if I better understood the available properties on a Section. The overview item now looks like this:
```html
      <li><a class="docs-link rounded{{ if $active }} active{{ end }}" href="/{{ .Section }}/">Overview</a></li>
```

I think there's only one small issue remaining, which is that I can't get the `active` CSS attribute assigned to the overview item in the list. I *think* this is because `.RelPermalink` evaluates to `section/subdirectory` like the above, where as my url is just `section/`. Given what I learned above, I currently have something like this, which I think is close but probably I'm doing string concatenation wrong:
```html
{{ $active := eq $currentPage.RelPermalink "/.Section/"}}
```

If you have any insight into a way to make the href a little cleaner or to fix the active attribute, it would be much appreciated. Thanks again!

**Comment by h-enk** [comment]
Sure!

![Snag_1dc72d53](https://user-images.githubusercontent.com/3902872/223962404-eb0c8470-b2f9-4530-bffc-d97b30b0e9c8.png)

`./layouts/partials/sidebar/auto-default-menu.html`:

```html
<!-- Auto default section menu -->
{{ $currentPage := . -}}
{{ $section := $currentPage.Section -}}
{{ range (where .Site.Sections "Section" "in" $section) }}
  {{ range .Sections }}
    {{ $active := in $currentPage.RelPermalink .RelPermalink }}
    <h3 class="h6 text-uppercase mb-2">{{ .Title }}</h3>
    <ul class="list-unstyled">
      <!-- start overview page -->
      {{ $active := eq $currentPage.RelPermalink .CurrentSection.RelPermalink }}
      <li><a class="docs-link rounded{{ if $active }} active{{ end }}" href="{{ .CurrentSection.RelPermalink }}">Overview</a></li>
      <!-- end overview page -->
      {{ range .Pages }}
        {{ if .IsNode }}
          {{ $active := in $currentPage.RelPermalink .RelPermalink }}
          <h4 class="h6 text-uppercase ms-3 mt-3 mb-2">{{ .Title }}</h4>
          <ul class="list-unstyled ms-3">
            {{ range .Pages }}
              {{ if .IsNode }}
                {{ $active := in $currentPage.RelPermalink .RelPermalink }}
                <h5 class="h6 text-uppercase mt-2 mb-2">{{ .Title }}</h5>
                <ul class="list-unstyled ms-3">
                  {{ range .Pages }}
                    {{ $active := in $currentPage.RelPermalink .RelPermalink }}
                    <li><a class="docs-link{{ if $active }} active{{ end }}" href="{{ .Permalink }}">{{ .Name }}</a></li>
                  {{ end }}
                </ul>
              {{ else }}
                {{ $active := in $currentPage.RelPermalink .RelPermalink }}
                <li><a class="docs-link{{ if $active }} active{{ end }}" href="{{ .Permalink }}">{{ .Name }}</a></li>
              {{ end }}
            {{ end }}
          </ul>
        {{ else }}
          {{ $active := in $currentPage.RelPermalink .RelPermalink }}
          <li><a class="docs-link{{ if $active }} active{{ end }}" href="{{ .Permalink }}">{{ .Name }}</a></li>
        {{ end }}
      {{ end }}
    </ul>
  {{ end }}
{{ end }}
```

Small UX fix, uncomment line 5 in `./assets/scss/common/_global.scss`:

```scss
.contributors .content,
.blog .content,
.page .content,
.error404 .content,
// .docs.list .content,
.tutorial.list .content,
.showcase.list .content,
.categories.list .content,
.tags.list .content {
  padding-top: 1rem;
  padding-bottom: 3rem;
}

[..]
```

**Reply by csparker247** [reply]
>       
>       {{ $active := eq $currentPage.RelPermalink .CurrentSection.RelPermalink }}
>       <li><a class="docs-link rounded{{ if $active }} active{{ end }}" href="{{ .CurrentSection.RelPermalink }}">Overview</a></li>
>       

Hmm, that still seems to be evaluating to the subdirectory inside my section (`new-imaging-methods/content/`) rather than just the section name (`new-imaging-methods/`). Since I haven't posted this, my section directory structure is setup like this:

``` shell
content/en/new-imaging-methods
├── _index.md                 # Overview page content (list type): First section, etc.
└── content
    ├── _index.md             # Just the section title in the front matter
    └── dark-field-imaging    # Second link in the section (single type)
        └── index.md  
```

> Small UX fix, uncomment line 5 in ./assets/scss/common/_global.scss:

Haha, you have a sharp eye! I had actually commented that out because the menu on the list page was noticeably lower than that of the single page:

<img width="596" alt="overview-page-compare" src="https://user-images.githubusercontent.com/1434526/224043999-7cc68137-ccca-42d2-bfc6-107dacb9e05f.png">

Since I want my list page to basically look like a single page, would you expect there to be any unintended consequences if I leave that commented out?

**Reply by h-enk** [reply]
> Since I want my list page to basically look like a single page, would you expect there to be any unintended consequences if I leave that commented out?

I expect no unintended consequences, so you could leave it commented out — I now see I wrote "uncomment" before, but I meant "comment out"

**Comment by csparker247** [comment]
I think what I want is violating the expected directory structure for sections and pages, so probably I'm just making more trouble for myself. 😄 

Anyway, I was able to get the `active` attribute to work by concatenating off of `.Section`:
```
      {{ $active := eq $currentPage.RelPermalink (print "/" .Section "/") }}
      <li><a class="docs-link rounded{{ if $active }} active{{ end }}" href="/{{ .Section }}/">Overview</a></li>
```

Since this is a pretty limited used site where all sections will be setup the same, I'm not going too much about this being a general solution.

Thanks for all of your help!
